geode-core/src/main/java/org/apache/geode/cache/client/internal/PoolImpl.java

```diff
@@ -227,6 +228,7 @@ protected PoolImpl(PoolManagerImpl pm, String name, Pool attributes,
           this.statisticInterval));
     }
     this.dsys = ds;
```

Comment:
It seems wrong to leave this code getting a InternalDistributedSystem by calling InternalDistributedSystem.getAnyInstance(). Now that you are passing a cache in you can call "getDistributedSystem()" on it.